File: course_101/exercises/medium_1/word_to_digit.rb
# Write a method that takes a sentence string as input, and returns the same string with any sequence of the words 'zero',
# 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine' converted to a string of digits.

STRING_TO_DIGIT = {
  'one' => '1', 'two' => '2', 'three' => '3',
  'four' => '4', 'five' => '5', 'six' => '6',
  'seven' => '7', 'eight' => '8', 'nine' => '9'
}

def word_to_digit(string)
  STRING_TO_DIGIT.keys.each do |word|
    string.gsub!(/\b#{word}\b/i, STRING_TO_DIGIT[word])
  end
  string
end

puts word_to_digit('Please call me at Five five five one Two three four. Thanks.') #== 'Please call me at 5 5 5 1 2 3 4. Thanks.'

File: course_js210_computational_thinking_and_javascript_programming/exercises/js210_small_problems/medium_problems/word_to_digit.js
// Write a function that takes a sentence string as an argument, and returns that string with every occurrence of a "number word" — 'zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine' — converted to its corresponding digit character.

// Further Exploration
// Can you refactor the function so that it does not use a loop?

const DIGIT_NOUNS = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];
const matchDigitNoun = /\b(one|two|three|four|five|six|seven|eight|nine)\b/gi;

const digitNounToDigit = function(noun) {
  return DIGIT_NOUNS.indexOf(noun);
}

function wordToDigit(sentence) {
  return sentence.replace(matchDigitNoun, digitNounToDigit);
}

console.log(wordToDigit('Please call me at five five five one two three four. Thanks. Height of the'));
// "Please call me at 5 5 5 1 2 3 4. Thanks."